WordPress/src/main/java/org/wordpress/android/ui/stats/StatsSinglePostDetailsActivity.java

```diff
@@ -0,0 +1,711 @@
+import android.support.v7.app.ActionBarActivity;
+import android.util.SparseBooleanArray;
+import android.view.LayoutInflater;
+import android.view.MenuItem;
+import android.view.View;
+import android.view.ViewGroup;
+import android.widget.BaseExpandableListAdapter;
+import android.widget.LinearLayout;
+import android.widget.TextView;
+
+import com.android.volley.VolleyError;
+import com.jjoe64.graphview.GraphView;
+import com.jjoe64.graphview.GraphViewSeries;
+import com.wordpress.rest.RestRequest;
+
+import org.json.JSONException;
+import org.json.JSONObject;
+import org.wordpress.android.R;
+import org.wordpress.android.WordPress;
+import org.wordpress.android.networking.RestClientUtils;
+import org.wordpress.android.ui.stats.models.PostViewsModel;
+import org.wordpress.android.ui.stats.models.SingleItemModel;
+import org.wordpress.android.ui.stats.models.VisitModel;
+import org.wordpress.android.util.AppLog;
+import org.wordpress.android.util.DisplayUtils;
+import org.wordpress.android.util.FormatUtils;
+import org.wordpress.android.util.NetworkUtils;
+import org.wordpress.android.util.ToastUtils;
+import org.wordpress.android.util.ptr.SwipeToRefreshHelper;
+
+import java.lang.ref.WeakReference;
+import java.util.List;
+import java.util.concurrent.Executors;
+import java.util.concurrent.ThreadPoolExecutor;
+
+
+/**
+ *  Single item details activity.
+ */
+public class StatsSinglePostDetailsActivity extends ActionBarActivity
+        implements StatsBarGraph.OnGestureListener{
+    public static final String ARG_REMOTE_POST_OBJECT = "ARG_REMOTE_POST_OBJECT";
+    private static final String ARG_REST_RESPONSE = "ARG_REST_RESPONSE";
+    private static final String ARG_SELECTED_GRAPH_BAR = "ARG_SELECTED_GRAPH_BAR";
+    private static final String UNDER_THE_GRAPH_DATE_LABEL_FORMAT = "MMMM d";
+
+    private boolean mIsUpdatingStats;
+    private SwipeToRefreshHelper mSwipeToRefreshHelper;
+
+    private final Handler mHandler = new Handler();
+
+    private LinearLayout mGraphContainer;
+    private TextView mStatsViewsLabel;
+    private TextView mStatsViewsTotals;
+    private LinearLayout mMonthsAndYearsList;
+    private LinearLayout mAveragesList;
+    private LinearLayout mRecentWeeksList;
+
+    private SingleItemModel mRemotePostItem; // The original item returned from TopPostsAndPages endpoint
+    private PostViewsModel mRestResponseParsed;
+    private int mSelectedBarGraphIndex = -1;
+
+    private SparseBooleanArray mYearsIdToExpandedMap;
+    private SparseBooleanArray mAveragesIdToExpandedMap;
+    private SparseBooleanArray mRecentWeeksIdToExpandedMap;
+
+    @Override
+    public void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+
+        setContentView(R.layout.stats_activity_single_post_details);
+
+        if (savedInstanceState == null) {
+            // AnalyticsTracker.track(AnalyticsTracker.Stat.STATS_ACCESSED);
+            // TODO: add analytics here
```

Comment:
This TODO along with [this one](https://github.com/wordpress-mobile/WordPress-Android/blob/issue/2153-stats-single-post-details-page/WordPress/src/main/java/org/wordpress/android/ui/stats/StatsViewAllActivity.java#L79) are pretty important - perhaps file a separate issue targeted for 3.7 to make sure these aren't overlooked?